### PR TITLE
feat: Working towards 1.0 release

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Big PR to clean up a lot of complex logic. Entirely removes `ethereumjs-vm`-related code. Removes `waffle`, `truffle`, and `ganache` tooling for now. Vastly simplifies compiler plugin by writing to a different artifacts folder instead of using `*-ovm.json`. Has the effect of also vastly simplifying every other plugin.

Remaining tasks:
- [ ] Update README to reflect changed plugins
- [x] Update package.json to remove old files

### CODE REVIEW

**Comment by platocrat** [comment]
If only there was a way to turn this PR into an NFT

**Comment by smartcontracts** [comment]
+309 −8,219

LOL

File: src/hardhat/compiler/index.ts [hunk 1]
```diff
@@ -130,49 +122,22 @@ subtask(
...
-              delete linkRefs[linkRefFileName][linkRefName]
-              linkRefs[linkRefFileName][`${linkRefName}-ovm`] = linkRefOutput
-            }
-          }
-
-          // OVM compiler output is signified by adding -ovm to the output name.
-          evmOutput.contracts[fileName][`${contractName}-ovm`] = contractOutput
-        }
-      }
-    }
```
Comment (gakonst) [inline src/hardhat/compiler/index.ts:168]: Note to reader: This can be removed because we're adding a new `artifacts-ovm` path to store them there.

**Review by gakonst — APPROVED** [review]
LGTM, my favorite kind of PR.

Let's add a note to the read me that the Artifacts will appear in a new `artifacts-ovm` folder, instead of how they were before (artifacts/{name}-ovm.json)

**Pull request merged**